File: rust/src/models/token_introspection_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Token introspection response as per RFC 7662
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct TokenIntrospectionResponse {
    pub active: bool,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub client_id: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub exp: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iat: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub scope: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub sub: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub token_type: Option<String>,
}

impl TokenIntrospectionResponse {
    pub fn new(active: bool) -> Self {
        Self {
            active,
            client_id: None,
            exp: None,
            iat: None,
            scope: None,
            sub: None,
            token_type: None,
        }
    }
}

File: typescript/src/models/tokenIntrospectionResponse.ts
// this file is @generated
/** Token introspection response as per RFC 7662 */
export interface TokenIntrospectionResponse {
  active: boolean;

  clientId?: string | null;

  exp?: number | null;

  iat?: number | null;

  scope?: string | null;

  sub?: string | null;

  tokenType?: string | null;
}

export const TokenIntrospectionResponseSerializer = {
  _fromJsonObject(object: any): TokenIntrospectionResponse {
    return {
      active: object["active"],
      clientId: object["client_id"],
      exp: object["exp"],
      iat: object["iat"],
      scope: object["scope"],
      sub: object["sub"],
      tokenType: object["token_type"],
    };
  },

  _toJsonObject(self: TokenIntrospectionResponse): any {
    return {
      active: self.active,
      client_id: self.clientId,
      exp: self.exp,
      iat: self.iat,
      scope: self.scope,
      sub: self.sub,
      token_type: self.tokenType,
    };
  },
};
